apps/cli-gui/src/app/guards/current-workspace.guard.ts

```diff
@@ -0,0 +1,51 @@
+import { HttpClient } from '@angular/common/http';
+import { inject } from '@angular/core';
+import { Router } from '@angular/router';
+import { CURRENT_WORKSPACE_PATH } from '@angular-cli-gui/shared/data';
+import { ConnectWorkspaceService } from '@angular-cli-gui/workspace-manager';
+import { catchError, map, Observable, of, retry, Subject, tap } from 'rxjs';
+
+import { CoreService } from '../core/core.service';
+
+export const currentWorkspaceGuard = (): Observable<boolean> => {
+  const router = inject(Router);
+  const http = inject(HttpClient);
+  const connectWorkspaceService = inject(ConnectWorkspaceService);
+  const core = inject(CoreService);
+  const retrySubject = new Subject<void>();
+  const projectNames$ = http.get<string[]>('/api/workspace');
+  const currentWorkspacePath = localStorage.getItem(CURRENT_WORKSPACE_PATH);
+
+  return projectNames$.pipe(
+    // Save projects to state
+    tap((projectNames) => {
+      core.update({
+        projectNames,
+        currentProjectName: projectNames?.[0],
+      });
+    }),
+    // Map to true to allow navigation
+    map(() => true),
+    catchError(() => {
+      if (!currentWorkspacePath) {
+        router.navigate(['workspace-manager', 'connect-workspace']);
+        return of(false);
+      }
+
+      // if path saved in local storage is invalid somehow (maybe deleted workspace)
+      // remove it from local storage so on 2nd retry cycle user will be navigated to workspace connection screen
+      localStorage.removeItem(CURRENT_WORKSPACE_PATH);
+
+      // Connect to workspace using local storage path and retry to get projectNames
+      return connectWorkspaceService
+        .connectWorkspace(currentWorkspacePath)
+        .pipe(
+          map(() => {
+            retrySubject.next();
```

Comment:
I appreciate the implementation but we need something that can be shared and reused, maybe we should consider adding ng-query

Reply:
I don't mind, but won't it make integrating this project with the angular repo more difficult? It might be a good idea to verify with them before relying on 3rd party libraries that aren't 100% necessary.